runbotics/runbotics-desktop/src/core/bpm/CustomServices.ts

```
@@ -36,118 +33,43 @@ export const customServices: Record<string, Function> = {
+    push: (array: any[], newElement: any) => {
+        if (!array) {
+            return [newElement];
         }
-        return arg1.concat([arg2]);
+        return array.concat([newElement]);
     },
     concat: (arg1: any, arg2: any) => {
         const array1 = arg1 ? arg1 : [];
         const array2 = arg2 ? arg2 : [];
 
         return [...array1, ...array2];
     },
-    splitAndPick: (arg1: any, pick: any) => {
-        console.log('executing splitAndPic', arg1, pick, 'BPM-service-call');
-        return arg1.split('-')[pick];
-    },
-    parseJson: (json: any) => {
-        console.info('parse json', json, 'BPM-service-call');
-        const res = JSON.parse(json);
-        console.info('parse json result', res, 'BPM-service-call');
-        return res;
+    split: (arg1: string, splitter: string) => {
+        return arg1.split(splitter);
     },
-
-    csvToObject: (cfg: any) => {
-        console.info('csvToObject - config', cfg, 'BPM-service-call');
-        const headers: string[] = cfg.csvHeader.split(',');
-        const content: string[] = cfg.csvContent.split(',');
-
-        const res = [];
-        let colIdx;
-        let row: any;
-        for (let idx = 0; idx < content.length; idx++) {
-            colIdx = idx % headers.length;
-
-            if (colIdx === 0) {
-                row = {};
-                res.push(row);
-            }
-
-            row[headers[colIdx]] = content[idx];
-        }
-        console.info('csvToObject - result', res, 'BPM-service-call');
-        return res;
-    },
-
-    comaSeparatedToArray: (s: any) => {
-        return s.split(',');
+    splitAndPick: (arg1: string, splitter: string, pick: number) => {
+        return arg1.split(splitter)[pick];
     },
-    withDefault: (def: any, val: any) => {
-        //console.log("with default:", def, val);
-        return val !== undefined && val !== null && val !== '' ? val : def;
+    parseJson: (json: any) => {
+        return JSON.parse(json);
```

parameter must be a string otherwise should throw an error